Authentication/IamTokenManager.cs

```diff
@@ -0,0 +1,186 @@
+/**
+* Copyright 2019 IBM Corp. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+
+using IBM.Cloud.SDK.Connection;
+using IBM.Cloud.SDK.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine.Networking;
+using Newtonsoft.Json;
+using System.Text;
+using Utility = IBM.Cloud.SDK.Utilities.Utility;
+
+
+namespace IBM.Cloud.SDK.Authentication
+{
+    public class IamTokenManager : JwtTokenManager
+    {
+        private string iamApikey;
+        private string iamClientId;
+        private string iamClientSecret;
+        private string iamDefaultUrl = "https://iam.cloud.ibm.com/identity/token";
+
+        private const string CLIENT_ID_SECRET_WARNING = "Warning: Client ID and Secret must BOTH be given, or the defaults will be used.";
+
+        public IamTokenManager(IamTokenOptions options) : base(options)
+        {
+			tokenName = "access_token";
```

Comment:
Are we using `tokenName` anywhere? I think this might be safe to remove from the TokenManagers since we are handling the json field name in the TokenData model.